### PR TITLE
docs: Feature Flags with Node.js and Unleash

### LINKED ISSUE #5290: docs: How to Implement Feature Flags in Node.js using Unleash
Outline 👇🏻

<hr>

# How to Implement Feature Flags in Node using Unleash

> Some additional information

In this tutorial, you will learn how to use feature flags in a [[Node.js](https://nodejs.org/)](https://nodejs.org/) application that displays onboarding flow to users, using Unleash. We will use the `unleash-client` package, which provides easy integration of Unleash feature flags in a Node.js application.

## What we’ll be using

Mention:
- [Tech](Link) (It's purpose)
- [Tech](Link) (It's purpose)
- [Tech](Link) (It's purpose)

## What you’ll need

- [Tech](Link) (It's purpose)
- [Tech](Link) (It's purpose)

## Setting up the project

To set up, just clone the app repo and follow this tutorial to learn everything that's in it. To fork the project, run:

```bash
git clone https://github.com/rishi-raj-jain/repo-slug
# Some additional information
```

## Scaffolding a Node.js app

Creating an Node.js app is as easy as a single command:

```bash
# command
```

## Setup Unleash

1. Run the following commands in the terminal to fetch the `docker-compose.yml` for creating an Unleash instance:

```bash
wget getunleash.io/docker-compose.yml
docker-compose up -d
```

This will start Unleash in the background. Once Unleash is running, you can access it at [http://localhost:4242/](http://localhost:4242/).

1. Now use the default credentials to log into the instance:

```bash
Username: admin
Password: unleash4all
```

## Create a New Feature

Create a new feature flag in your Unleash instance named `user-access`.

### Installation

To get started with Node.js and Unleash, you need to install `unleash-client` package as a dependency.

You can run the following commands in your terminal to do this:

```bash
npm install unleash-client
```

### Set up Environment Variables

By default, the following values are setup in your local Unleash instance

```bash
# .env

UNLEASH_API_URL="http://localhost:4242/api"
UNLEASH_AUTHORIZATION_KEY="default:development.unleash-insecure-api-token"
```

### Initialize Unleash SDK

> Some additional information

### Create a simple HTTP server with Node.js

> Some additional information

```typescript
// some code
```

### Use Unleash SDK to fetch the feature flag value

> Some additional information

```typescript
// some code
```

## **Scenarios**

> Some additional information

> Screenshots

## Using Unleash in Production

To setup Unleash for production, please follow the steps below:

1. Self-host Unleash, or run an instance on [[Unleash Cloud](https://www.getunleash.io/pricing)](https://www.getunleash.io/pricing).
2. Get an [[API key](https://github.com/reference/api-tokens-and-client-keys)](/reference/api-tokens-and-client-keys) from the Unleash dashboard.
3. Store the API key in your Environment Variables of your hosting, which secures it and makes it accessible in your code.

## Conclusion

> Some additional information

### PR DESCRIPTION
**Pull request by rishi-raj-jain**
resolve #5290 

This PR aims to illustrate the implementation of Feature Flags using Unleash and Node.js

- The best thing about this is readers get to learn basic `http` server (and not express)
- The app that's being referenced in the docs: https://github.com/rishi-raj-jain/user-access-feature-flag-with-unleash-and-node
- I've added beautiful screenshots via https://screenshot-beautifier.vercel.app/
- I've used the best practices in the code using res object of basic `http` server

P.S. My second contribution as an Unleash ambassador!

cc @pransh15

### CODE REVIEW

File: website/docs/feature-flag-tutorials/node/implementing-feature-flags.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+    // This is just an example, but you might want to persist user data
+    // through cookies on the server and not rely on the request body
+    // to send it to the server for security reasons
+    // ...
+    
+    // Access the parsed form fields using the 'fields' object from formidable
+    const userId = fields.email[0]
+    
+    // Load Unleash SDK
+    const unleash = await getUnleash()
```
Comment (ivarconr) [inline website/docs/feature-flag-tutorials/node/implementing-feature-flags.md:165]: This promotes an incorrect way to use Unleash. Unleash should be initialized outside of the context of the "request scope" of the application. 

Also highlighted in this comment: https://github.com/Unleash/unleash/pull/5246/files#r1385619950

Comment (rishi-raj-jain) [reply website/docs/feature-flag-tutorials/node/implementing-feature-flags.md:165]: @ivarconr 

Thanks for teaching me that. I'm learning 😁

I've updated the docs and the git repo with that change.

**Comment by rishi-raj-jain** [comment]
The updates that I've just done, addresses two important points:

- Breaking the whole tutorial into smaller parts: the nextjs guide right now lacks hierarchy, and both my PRs aim to get a lot better at this.

- `Revert` section to the guides: This is instant demonstration of how users can take out feature flags in their application without needing to re-deploy the code.

**Comment by rishi-raj-jain** [comment]
@ivarconr

I've made the changes, fixing the issues you pointed out.

Do let me know if anything.

**Review by ivarconr — APPROVED** [review]
I think this looks great now. Easy to follow for node.js folks new to Unleash! 

Thanks for the effort @rishi-raj-jain

**Comment by rishi-raj-jain** [comment]
Published!

https://dev.to/reeshee/how-to-implement-feature-flags-in-nodejs-using-unleash-3907

Social Media 👇🏻

- https://twitter.com/rishi_raj_jain_/status/1741904886097387683
- https://www.linkedin.com/feed/update/urn:li:activity:7147670342453407745/
- https://peerlist.io/scroll/post/ACTHOK77DRDJ9QKDE37E6DNKJB9BJO

**Comment by rishi-raj-jain** [comment]
Updated the GitHub Repo with a README:

https://github.com/rishi-raj-jain/user-access-feature-flag-with-unleash-and-node/tree/master

**Comment by rishi-raj-jain** [comment]
Updated the GitHub Repo with a README:

https://github.com/rishi-raj-jain/user-access-feature-flag-with-unleash-and-node

**Review by dgorton — COMMENTED** [review]
This is great. Thanks again for putting this together. Going to close this PR as it's been published at Dev.to

**Pull request closed without merging**